ocs_ci/helpers/vdbench_helpers.py

```diff
@@ -0,0 +1,622 @@
+            report["performance"]["avg_rate"]
+        ) / len(report["performance"]["avg_rate"])
+
+    if report["performance"]["response_times"]:
+        report["performance"]["average_response_time"] = sum(
+            report["performance"]["response_times"]
+        ) / len(report["performance"]["response_times"])
+
+    # Save to file if requested
+    if output_file:
+        try:
+            with open(output_file, "w") as f:
+                yaml.dump(report, f, default_flow_style=False)
+            log.info(f"Performance report saved to: {output_file}")
+        except Exception as e:
+            log.error(f"Failed to save report: {e}")
+
+    return report
+
+
+def cleanup_vdbench_resources(namespace, deployment_name):
+    """
+    Clean up Vdbench resources manually.
+
+    Args:
+        namespace: Kubernetes namespace
+        deployment_name: Name of the deployment to clean up
+    """
+    resources = [f"deployment/{deployment_name}", f"configmap/{deployment_name}-config"]
+
+    for resource in resources:
+        try:
+            run_cmd(f"oc delete {resource} -n {namespace} --ignore-not-found=true")
+            log.info(f"Deleted {resource}")
+        except Exception as e:
+            log.warning(f"Failed to delete {resource}: {e}")
+
+
+def wait_for_vdbench_pods_ready(namespace, deployment_name, timeout=300):
+    """
+    Wait for Vdbench pods to be ready.
+
+    Args:
+        namespace: Kubernetes namespace
+        deployment_name: Name of the deployment
+        timeout: Timeout in seconds
+
+    Returns:
+        bool: True if pods are ready, False if timeout
+    """
+    from ocs_ci.utility.utils import TimeoutSampler
+
+    try:
+        for sample in TimeoutSampler(
+            timeout, 10, _check_pods_ready, namespace, deployment_name
+        ):
+            if sample:
+                return True
+        return False
+    except Exception as e:
+        log.error(f"Error waiting for pods: {e}")
+        return False
+
+
+def _check_pods_ready(namespace, deployment_name):
+    """
+    Check if pods are ready.
+
+    Args:
+        namespace: Kubernetes namespace
+        deployment_name: Name of the deployment
+
+    Returns:
+        bool: True if all pods are ready
+    """
+    try:
+        cmd = (
+            f"oc get pods -n {namespace} -l app={deployment_name} "
+            f"-o jsonpath='{{.items[*].status.conditions[?(@.type==\"Ready\")].status}}'"
+        )
+        result = run_cmd(cmd)
+        ready_statuses = result.strip().split()
+        return (
+            all(status == "True" for status in ready_statuses)
+            and len(ready_statuses) > 0
+        )
+    except Exception:
+        return False
+
+
+def validate_vdbench_workload_health(workload, timeout=300):
+    """
+    Validate that a Vdbench workload is healthy and running.
+    Args:
+        workload: VdbenchWorkload instance to validate
+        timeout: Timeout for validation in seconds
+    Returns:
```

Comment:
The function returns a bool value but only True and never False. 
How about returning False instead of rasing AssertionError?

Reply:
Yes, agreed. I have updated the code to return False instead of raising an assertion.